common/models/facility.js

```diff
@@ -1,5 +1,266 @@
+								}
+							}
+						]
+					};
+					var pdfDoc = printer.createPdfKitDocument(docDefinition);
+					var pdfFile = pdfDoc.pipe(
+						fs.createWriteStream("facilities.pdf")
+					);
+					pdfDoc.end();
+				}
+
+				if (inputData.format == "excel") {
+					const jsonArrayData = [];
+					queriedDetails.forEach(details => {
+						const jsonDetails = details.toJSON();
+						jsonArrayData.push({
+							CODE: jsonDetails.facility_code,
+							NAME: jsonDetails.facility_name,
+							"COMMON NAME": jsonDetails.facility_name,
+							OWNERSHIP: jsonDetails.owner.facility_owner,
+							TYPE: jsonDetails.facilityType.facility_type,
+							STATUS:
+								jsonDetails.operationalStatus
+									.facility_operational_status,
+							ZONE: jsonDetails.district.zone.zone_name,
+							DISTRICT: jsonDetails.district.district_name,
+							"DATE OPENED": moment(
+								jsonDetails.facility_date_opened
+							).format("MMM Do YY"),
+						});
+					});
+
+					res.xls("data.xlsx", jsonArrayData);
+
+					// fs.writeFileSync("facilities.xlsx", xlsDetails, "binary");
+				}
+
+				if (inputData.format == "csv") {
+					var fields = [
+						"CODE",
+						"NAME",
+						"COMMON NAME",
+						"OWNERSHIP",
+						"TYPE",
+						"STATUS",
+						"ZONE",
+						"DISTRICT",
+						"DATE OPENED",
+					];
+					const csvArrayData = [];
+					queriedDetails.forEach(details => {
+						const jsonDetails = details.toJSON();
+						csvArrayData.push({
+							CODE: jsonDetails.facility_code,
+							NAME: jsonDetails.facility_name,
+							"COMMON NAME": jsonDetails.facility_name,
+							OWNERSHIP: jsonDetails.owner.facility_owner,
+							TYPE: jsonDetails.facilityType.facility_type,
+							STATUS:
+								jsonDetails.operationalStatus
+									.facility_operational_status,
+							ZONE: jsonDetails.district.zone.zone_name,
+							DISTRICT: jsonDetails.district.district_name,
+							"DATE OPENED": moment(
+								jsonDetails.facility_date_opened
+							).format("MMM Do YY"),
+						});
+					});
+					var csvDetails = json2csv({
+						data: csvArrayData,
+						fields: fields,
+					});
+					const datetime = new Date();
+					res.set("Expires", "Tue, 03 Jul 2001 06:00:00 GMT");
+					res.set(
+						"Cache-Control",
+						"max-age=0, no-cache, must-revalidate, proxy-revalidate"
+					);
+					res.set("Last-Modified", datetime + "GMT");
+					res.set("Content-Type", "application/force-download");
+					res.set("Content-Type", "application/octet-stream");
+					res.set(
+						"Content-Disposition",
+						"attachment;filename=facilities.csv"
+					);
+					res.set("Content-Transfer-Encoding", "binary");
+					res.send(csvDetails);
+				}
+			}
+		);
+	};
 
+	Facility.remoteMethod("download", {
+		description: "Download all facilities of your choice",
+		accepts: [
+			{arg: "data", type: "object"},
+			{arg: "res", type: "object", http: {source: "res"}}
+		],
+		returns: {}
+	});
+	Facility.downloadOne = function(inputData, res, cb) {
```

Comment:
Find out in the loopback documentation if you can have two remote methods of the same model in two separate files